addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_materials.py

```diff
@@ -136,10 +144,15 @@ def __gather_extensions(blender_material, export_settings):
         if blender_material.use_shadeless:
             extensions["KHR_materials_unlit"] = Extension("KHR_materials_unlit", {}, False)
     else:
-        if gltf2_blender_get.get_socket_or_texture_slot(blender_material, "Background") is not None:
-            extensions["KHR_materials_unlit"] = Extension("KHR_materials_unlit", {}, False)
-
-    # TODO specular glossiness extension
+        if blender_material.node_tree and blender_material.use_nodes:
+            # Materials using at least one Background or Emissive node, without a
+            # Principled BSDF node, should be considered Unlit.
+            # See: https://github.com/KhronosGroup/glTF-Blender-IO/issues/554
+            has_background = __has_node_of_type(blender_material, bpy.types.ShaderNodeBackground)
+            has_emission = __has_node_of_type(blender_material, bpy.types.ShaderNodeEmission)
+            has_principled = __has_node_of_type(blender_material, bpy.types.ShaderNodeBsdfPrincipled)
+            if not has_principled and (has_background or has_emission):
```

Comment:
This seems a more robust check.  Perhaps this could be encapsulated somewhere common?